### PR TITLE
add Kubernetes

### PR DESCRIPTION
**Pull request by Liang1XZhao**
Signed-off-by: Liang1XZhao <liang1x.zhao@intel.com>

### CODE REVIEW

File: ehsm_kms_service/kubernetes/readme.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        cat > /etc/exports <<EOF 
+        /intel-ehsm-db *(rw,no_root_squash,no_all_squash,sync) 
+        EOF
+        
+        systemctl start rpcbind & systemctl enable rpcbind
+        systemctl start nfs & systemctl enable nfs
+        
+    2. All Kubernetes node
+    
+        yum -y install nfs-utils recbind
```
Comment (yang8621) [inline ehsm_kms_service/kubernetes/readme.md:19]: recbind ?

File: ehsm_kms_service/kubernetes/readme.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        mkdir /intel-ehsm-db
+        chmod 777 /intel-ehsm-db
+        chown nfsnobody /intel-ehsm-db
+        
+        cat > /etc/exports <<EOF 
+        /intel-ehsm-db *(rw,no_root_squash,no_all_squash,sync) 
+        EOF
+        
+        systemctl start rpcbind & systemctl enable rpcbind
+        systemctl start nfs & systemctl enable nfs
```
Comment (yang8621) [inline ehsm_kms_service/kubernetes/readme.md:15]: && ?

File: ehsm_kms_service/kubernetes/readme.md (hunk 1 above)
Comment (yang8621) [reply ehsm_kms_service/kubernetes/readme.md:19]: not sure whether need to start rpcbind here?

Comment (Liang1XZhao) [reply ehsm_kms_service/kubernetes/readme.md:19]: typo, rpcbind.

Comment (Liang1XZhao) [reply ehsm_kms_service/kubernetes/readme.md:19]: k8s pv needs it

File: ehsm_kms_service/kubernetes/readme.md (hunk 2 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/kubernetes/readme.md:15]: yes

**Comment by yang8621** [comment]
Seems the previous comments are not fixed.

File: ehsm_kms_service/kubernetes/ehsm-pv.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: PersistentVolume 
+metadata:
+  name: nfspv-ehsm
+spec:
+  capacity:
+    storage: 1Gi
```
Comment (Liang1XZhao) [inline ehsm_kms_service/kubernetes/ehsm-pv.yaml:7]: change to 10Gi

**Review by syan10 — COMMENTED** [review]
How do you think if we combine all the yaml file into a one file (e.g. ehsm-kms.yaml)?

File: ehsm_kms_service/kubernetes/ehsm-db.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: couchdb
+  namespace: intel-ehsm
```
Comment (syan10) [inline ehsm_kms_service/kubernetes/ehsm-db.yaml:5]: can we change the name to "ehsm-kms", don't want to use the intel-xxx

File: ehsm_kms_service/kubernetes/ehsm-pv.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  name: nfspv-ehsm
+spec:
+  capacity:
+    storage: 1Gi 
+  accessModes:
+    - ReadWriteOnce 
+  persistentVolumeReclaimPolicy: Retain 
+  storageClassName: nfs
+  nfs:
+    path: /nfs_ehsm_db
```
Comment (syan10) [inline ehsm_kms_service/kubernetes/ehsm-pv.yaml:13]: please keep consistent naming rules (e.g ehsm_xxx)

File: ehsm_kms_service/kubernetes/readme.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
+Step 1 Installing the NFS Server
```
Comment (syan10) [inline ehsm_kms_service/kubernetes/readme.md:1]: this file seem don't need anymore,  or you can just add a link to directly point the details to deployment_instructions.md

File: ehsm_kms_service/kubernetes/ehsm-db.yaml (hunk 4 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/kubernetes/ehsm-db.yaml:5]: It will be change

File: ehsm_kms_service/kubernetes/ehsm-pv.yaml (hunk 5 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/kubernetes/ehsm-pv.yaml:13]: It will be fix.

File: ehsm_kms_service/kubernetes/ehsm-kms.yaml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        - name: EHSM_CONFIG_COUCHDB_PORT
+          valueFrom:
+            configMapKeyRef:
+              name: ehsm-configmap
+              key: database_port
+        - name: EHSM_CONFIG_COUCHDB_DB
+          valueFrom:
+            configMapKeyRef:
+              name: ehsm-configmap
+              key: database_name
```
Comment (syan10) [inline ehsm_kms_service/kubernetes/ehsm-kms.yaml:58]: can we combine the  the env variable into one?
EHSM_CONFIG_COUCHDB_DB_URL?

which the EHSM_CONFIG_COUCHDB_DB_URL=http://couch-root-username:couch-root-password@database_url:database_port?

Comment (Liang1XZhao) [reply ehsm_kms_service/kubernetes/ehsm-kms.yaml:58]: It's better to set it separately

File: ehsm_kms_service/kubernetes/readme.md (hunk 6 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/kubernetes/readme.md:1]: It will be remove.

**Pull request closed without merging**